### ISSUE TITLE
`jsr:@std/http/file-server` stucks the compiled programs.

### ISSUE THREAD

**Issue by Dreagonmon** [denoland/deno_std#4658]
When using the `serveDir` from `jsr:@std/http/file-server`, and compile the script using `deno compile`, the compiled program won`t start running.

**Steps to Reproduce**

main.ts
```ts
console.log("before everything");
import { serveDir } from "@std/http/file-server";
const handler: Deno.ServeHandler = async (req, _info) => {
  // static server
  return serveDir(req, {
    fsRoot: "./",
  });
};
const __main__ = async () => {
  // start server
  const server = Deno.serve({
    hostname: "0.0.0.0",
    port: 8000,
  }, handler);
};
// Learn more at https://deno.land/manual/examples/module_metadata#concepts
if (import.meta.main) {
  await __main__();
}
```

deno.json
```json
{
  "tasks": {
    "dev": "deno run --watch --allow-read --allow-net main.ts",
    "compile-run": "deno compile --allow-read --allow-net --output main_exec main.ts && ./main_exec"
  },
  "imports": {
    "@std/http/file-server": "jsr:@std/http@^0.224.0/file-server"
  }
}
```

Compile and run: `deno task compile-run`

And the output is:
```
Task compile-run deno compile --allow-read --allow-net --output main_exec main.ts && ./main_exec
Check file:///home/dreagonmon/Data/Temp/deno_test/main.ts
Compile file:///home/dreagonmon/Data/Temp/deno_test/main.ts to main_exec
Archive:  /tmp/.tmpk2MG2L/denort.zip
  inflating: denort

```

And it stucks here, no console output, the http server also not started.

**Expected behavior**

Change the import map:
```json
{
  "tasks": {
    "dev": "deno run --watch --allow-read --allow-net main.ts",
    "compile-run": "deno compile --allow-read --allow-net --output main_exec main.ts && ./main_exec"
  },
  "imports": {
    "@std/http/file-server": "https://deno.land/std@0.224.0/http/file_server.ts"
  }
}
```

Compile and run: `deno task compile-run`

The output is:
```
Task compile-run deno compile --allow-read --allow-net --output main_exec main.ts && ./main_exec
Check file:///home/dreagonmon/Data/Temp/deno_test/main.ts
Compile file:///home/dreagonmon/Data/Temp/deno_test/main.ts to main_exec
Archive:  /tmp/.tmpKDsYIB/denort.zip
  inflating: denort                  
before everything
Listening on http://localhost:8000/
```

**Environment**

- OS: Linux 6.6.26-1-MANJARO
- deno version: deno 1.42.4 (release, x86_64-unknown-linux-gnu)
- std version: 0.224.0

**Comment by iuioiua** [comment]
I replicated this issue on my macOS machine when using Deno 1.42.4. However, the problem appears fixed when using Deno canary, meaning bug fixes will soon be released in 1.43.0. Can you please upgrade to Deno canary and test again?

**Comment by kt3k** [comment]
probably the same issue as https://github.com/denoland/deno/issues/23551 and https://github.com/denoland/deno/issues/23139

resolved in https://github.com/denoland/deno/pull/23567

**Comment by Dreagonmon** [comment]
> Can you please upgrade to Deno canary, using `deno upgrade --canary`, and test again?

Yes, the new version seems fixed this bug, 1.43.0(canary) works as expected.
